gateway/README.md

```diff
@@ -11,158 +11,269 @@ This basic example demonstrates how to expose a service or a function through an
+    ```
+
+2. Fetch the access token with required scopes. The access token in the response is referred to as the **\{oauth2-token\}**. Run:
+
+    ```bash
+    curl https://oauth2.$KYMA_EXAMPLE_DOMAIN/oauth2/token -H "Authorization: Basic ZXhhbXBsZS1pZDpleGFtcGxlLXNlY3JldA==" -F "grant_type=client_credentials" -F "scope=read write"
+    ```
+
+#### Test the APIs with OAuth2 authentication
+
+```bash
+# To perform a test without the token, use the following command:
+curl -ik https://http-db-service.$KYMA_EXAMPLE_DOMAIN/orders
+# > {"error":{"code":401,"status":"Unauthorized","request":"530f300a-8269-4564-8d0c-9816c692e7c4","message":"The request could not be authorized"}}
+
+# To perform a test with the token, use the following command:
+curl -ik https://http-db-service.$KYMA_EXAMPLE_DOMAIN/orders -H 'Authorization: Bearer {oauth2-token}'
+# > 200 []
 ```
 
 ### Cleanup
 
 Run the following command to completely remove the example and all its resources from the cluster:
 
 ```bash
-kubectl delete all -l example=gateway-lambda -n $KYMA_EXAMPLE_NS
 kubectl delete all -l example=gateway-service -n $KYMA_EXAMPLE_NS
+kubectl delete apirules.gateway.kyma-project.io -l example=gateway-service -n $KYMA_EXAMPLE_NS
+kubectl delete oauth2clients.hydra.ory.sh -l example=gateway-service -n $KYMA_EXAMPLE_NS
 ```
 
 ## Troubleshooting
 
-The problem occurs when there is an Api resource with authentication enabled, but after making a request without a JWT token, the received response code is `200`.
+**Could not resolve host:** If you run Kyma locally, make sure you have added the hostnames used in this example to your hosts file.
 
-**Solution 1:** Wait. If the cluster is under high workload, it can take a while for authentication policies to apply. If you still have the problem after a few seconds, look at the Solution 2.
+**No healthy upstream:** Check if the Pod you created is running. Run:
 
-**Solution 2:** If you did not use the default settings, there can be something wrong with the JWKS URI you provided. If you use a local Deployment of Kyma on Minikube and the internal OIDC Identity Provider such as Dex, make sure that the JWKS URI is provided as FQDN, and that it points directly to the keys endpoint, for example, http://dex-service.kyma-system.svc.cluster.local:5556/keys. Envoy sidecars must be able to resolve a domain name to the proper inside-cluster or outside-cluster IP address.
+```bash
+kubectl get pods -n $KYMA_EXAMPLE_NS
+```
 
-**Solution 3:** Check if the Pod you created has the istio-proxy container injected. Run this command:
+Wait until all containers of the Pod are running.
+
+**Problems with the JWT authentication:** Make sure you have provided proper domain name in the **Expose a service with JWT authentication** step.
+
+**Upstream connect error or disconnect/reset before headers:** Check if the Pod you created has the istio-proxy container injected. Run this command:
 
 ```bash
 kubectl get pods -n $KYMA_EXAMPLE_NS
 ```
 
```

Comment:
We can get rid of the fist command completely
```suggestion
**Upstream connect error or disconnect/reset before headers:** Check if the Pod you created has the istio-proxy container injected. Run this command:

```bash
kubectl get pods -n $KYMA_EXAMPLE_NS
```
**Upstream connect error or disconnect/reset before headers:** Check if the Pod you created has the istio-proxy container injected.
```